ts/features/bonus/bpd/components/optInStatus/BottomSheetMethodsToDelete.tsx

```diff
@@ -0,0 +1,111 @@
+import { BlockButtonProps } from "../../../../../components/ui/BlockButtons";
+import { H3 } from "../../../../../components/core/typography/H3";
+import { IOStyles } from "../../../../../components/core/variables/IOStyles";
+
+type Props = {
+  onDeletePress: () => void;
+  onCancelPress?: () => void;
+  paymentMethods: ReadonlyArray<PaymentMethod>;
+};
+
+export const BottomSheetMethodsToDelete = (props: Props) => {
+  const deleteProps: BlockButtonProps = {
+    testID: "deleteButtonTestID",
+    style: {
+      flex: 1,
+      borderColor: IOColors.red
+    },
+    onPressWithGestureHandler: true,
+    labelColor: IOColors.red,
+    bordered: true,
+    onPress: props.onDeletePress,
+    title: I18n.t("global.buttons.delete")
+  };
+  const cancelProps: BlockButtonProps = {
+    testID: "cancelButtonTestID",
+    bordered: true,
+    onPressWithGestureHandler: true,
+    onPress: props.onCancelPress,
+    title: I18n.t("global.buttons.cancel")
+  };
+  return (
+    <BottomSheetContent
+      testID={"BottomSheetMethodsToDeleteTestID"}
+      footer={
+        <FooterWithButtons
+          type={"TwoButtonsInlineHalf"}
+          leftButton={deleteProps}
+          rightButton={cancelProps}
+        />
+      }
+    >
+      <Label color={"bluegrey"} weight={"Regular"}>
+        {I18n.t(
+          "bonus.bpd.optInPaymentMethods.deletePaymentMethodsBottomSheet.subtitle"
+        )}
+      </Label>
+      <View spacer={true} />
+      {props.paymentMethods.map(pm => (
+        <ListItem
+          key={`payment_method_${pm.idWallet}`}
+          testID={`payment_method_${pm.idWallet}`}
+        >
+          <PaymentMethodRepresentationComponent {...pm} />
+        </ListItem>
+      ))}
+    </BottomSheetContent>
+  );
+};
+
+type BottomSheetReturnType = {
+  presentBottomSheet: () => void;
+};
+
+/**
+ * return an hook that exposes presentBottomSheet function to imperative show a bottomsheet
+ * containing the list of those payment methods that have BPD as function enabled
+ * @param props
+ */
+export const useBottomSheetMethodsToDelete = (
+  props: Pick<Props, "onDeletePress" | "onCancelPress">
+): BottomSheetReturnType => {
+  const paymentMethods = useSelector(getBPDMethodsSelector);
+  const snapPoint = Math.min(
+    Dimensions.get("window").height * 0.8,
+    // (subtitle + footer) + items
+    280 + paymentMethods.length * 58
+  );
+  const { present } = useIOBottomSheetRaw(snapPoint);
+  return {
+    presentBottomSheet: () => {
+      void present(
+        <BottomSheetMethodsToDelete
+          paymentMethods={paymentMethods}
+          onDeletePress={props.onDeletePress}
+          onCancelPress={props.onCancelPress}
+        />,
+        <View style={IOStyles.flex}>
+          <H3>
+            {I18n.t(
+              "bonus.bpd.optInPaymentMethods.deletePaymentMethodsBottomSheet.title"
+            )}
+          </H3>
+          <View spacer={true} />
+        </View>
+      );
+    }
+  };
+};
```

Comment:
```suggestion
export const useBottomSheetMethodsToDelete = (
  props: Pick<Props, "onDeletePress" | "onCancelPress">
): BottomSheetReturnType => {
  const paymentMethods = useSelector(getBPDMethodsSelector);
  const snapPoint = Math.min(
    Dimensions.get("window").height * 0.8,
    // (subtitle + footer) + items
    280 + paymentMethods.length * 58
  );
  
  const bottomSheetContent = (
    <BottomSheetMethodsToDelete
      paymentMethods={paymentMethods}
      onDeletePress={props.onDeletePress}
      onCancelPress={props.onCancelPress}
    />
  );

  const bottomSheetHeader = (
    <View style={IOStyles.flex}>
      <H3>
        {I18n.t(
          "bonus.bpd.optInPaymentMethods.deletePaymentMethodsBottomSheet.title"
        )}
      </H3>
      <View spacer={true} />
    </View>
  );

  const { present: presentBottomSheet } = useIOBottomSheet(
    bottomSheetContent,
    bottomSheetHeader,
    snapPoint
  );

  return {
    presentBottomSheet
  };
};
```
Just a suggestion, mainly helpful for the bottom-sheet library update. In this case the usage of `useIOBottomSheet` is enough what do you think? (didn't tested it)

Reply:
I guess it doesn't work.
I'm pretty sure I tried your approach and I found some down sides
This is what happen

### your suggestion
https://user-images.githubusercontent.com/822471/153578742-fa88a832-2fd5-419c-aa8c-61e5779d76e9.mov

### current
https://user-images.githubusercontent.com/822471/153578804-5eaaddfe-8f12-454b-b60b-768a7c7c1b90.mov

Reply:
not at the moment, but the footer is integrated in [`react-native-bottom-sheet` v4](https://gorhom.github.io/react-native-bottom-sheet/components/bottomsheetfooter)

We will solve even this issue 😄